pkg/chunk/openstack/swift_object_client.go

```diff
@@ -0,0 +1,207 @@
+	UserId                      string        `yaml:"user_id"`               // User Id
+	ApiKey                      string        `yaml:"api_key"`               // Key for api access
+	ApplicationCredentialId     string        `yaml:"app_credential_id"`     // Application Credential ID
+	ApplicationCredentialName   string        `yaml:"app_crendential_name"`  // Application Credential Name
+	ApplicationCredentialSecret string        `yaml:"app_credential_secret"` // Application Credential Secret
+	AuthUrl                     string        `yaml:"auth_url"`              // Auth URL
+	Retries                     int           `yaml:"retries"`               // Retries on error (default is 3)
+	ConnectTimeout              time.Duration `yaml:"connect_timeout"`       // Connect channel timeout (default 10s)
+	Timeout                     time.Duration `yaml:"timeout"`               // Data channel timeout (default 60s)
+	Region                      string        `yaml:"region"`                // Region to use eg "LON", "ORD" - default is use first region (v2,v3 auth only)
+	AuthVersion                 int           `yaml:"auth"`                  // Set to 1, 2 or 3 or leave at 0 for autodetect
+	Internal                    bool          `yaml:"internal"`              // Set this to true to use the the internal / service network
+	Tenant                      string        `yaml:"tenant"`                // Name of the tenant (v2,v3 auth only)
+	TenantId                    string        `yaml:"tenant_id"`             // Id of the tenant (v2,v3 auth only)
+	EndpointType                string        `yaml:"endpoint_type"`         // Endpoint type (v2,v3 auth only) (default is public URL unless Internal is set)
+	TenantDomain                string        `yaml:"tenant_domain"`         // Name of the tenant's domain (v3 auth only), only needed if it differs from the user domain
+	TenantDomainId              string        `yaml:"tenant_domain_id"`      // Id of the tenant's domain (v3 auth only), only needed if it differs the from user domain
+	TrustId                     string        `yaml:"trust_id"`              // Id of the trust (v3 auth only)
+}
+
+// RegisterFlags registers flags.
+func (cfg *SwitfConfig) RegisterFlags(f *flag.FlagSet) {
+	cfg.RegisterFlagsWithPrefix("", f)
+}
+
+// RegisterFlagsWithPrefix registers flags with prefix.
+func (cfg *SwitfConfig) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
+	f.StringVar(&cfg.ContainerName, prefix+"swift.container-name", "cortex", "Name of the Swift container to put chunks in.")
+	f.StringVar(&cfg.Domain, prefix+"swift.domain", "", "Openstack user's domain name.")
+	f.StringVar(&cfg.DomainId, prefix+"swift.domain-id", "", "Openstack user's domain id.")
+	f.StringVar(&cfg.UserName, prefix+"swift.username", "", "Openstack username for the api.")
+	f.StringVar(&cfg.UserId, prefix+"swift.userid", "", "Openstack userid for the api.")
```

Comment:
Just to keep it consistent with the YAML:

```suggestion
	f.StringVar(&cfg.UserName, prefix+"swift.user-name", "", "Openstack username for the api.")
	f.StringVar(&cfg.UserId, prefix+"swift.user-id", "", "Openstack userid for the api.")
```